quantum/repeat_key.c

```diff
@@ -0,0 +1,282 @@
+        }
+
+        if (!alt_keycode) {
+            // The following key pairs are considered with any mods.
+            // clang-format off
+            static const uint8_t pairs[][2] PROGMEM = {
+                {KC_LEFT, KC_RGHT},  // Left / Right Arrow.
+                {KC_UP  , KC_DOWN},  // Up / Down Arrow.
+                {KC_HOME, KC_END },  // Home / End.
+                {KC_PGUP, KC_PGDN},  // Page Up / Page Down.
+                {KC_BSPC, KC_DEL },  // Backspace / Delete.
+                {KC_LBRC, KC_RBRC},  // Brackets [ ] and { }.
+#ifdef EXTRAKEY_ENABLE
+                {KC_WBAK, KC_WFWD},  // Browser Back / Forward.
+                {KC_MNXT, KC_MPRV},  // Next / Previous Media Track.
+                {KC_MFFD, KC_MRWD},  // Fast Forward / Rewind Media.
+                {KC_VOLU, KC_VOLD},  // Volume Up / Down.
+                {KC_BRIU, KC_BRID},  // Brightness Up / Down.
+#endif  // EXTRAKEY_ENABLE
+#ifdef MOUSEKEY_ENABLE
+                {KC_MS_L, KC_MS_R},  // Mouse Cursor Left / Right.
+                {KC_MS_U, KC_MS_D},  // Mouse Cursor Up / Down.
+                {KC_WH_L, KC_WH_R},  // Mouse Wheel Left / Right.
+                {KC_WH_U, KC_WH_D},  // Mouse Wheel Up / Down.
+#endif  // MOUSEKEY_ENABLE
+            };
+            // clang-format on
+            alt_keycode = find_alt_keycode(pairs, sizeof(pairs), keycode);
+        }
+
+        if (alt_keycode) {
+            // Combine basic keycode with mods.
+            return (mods << 8) | alt_keycode;
+        }
+    }
+
+    return KC_NO; // No alternate key found.
+}
+
+void alt_repeat_key_invoke(const keyevent_t* event) {
+    static keyrecord_t registered_record       = {0};
+    static int8_t      registered_repeat_count = 0;
+    // Since this function calls process_record(), it may recursively call
+    // itself. We return early if `processing_repeat_count` is nonzero to
+    // prevent infinite recursion.
+    if (processing_repeat_count) {
+        return;
+    }
+
+    if (event->pressed) {
+        registered_record = (keyrecord_t){
+#    ifndef NO_ACTION_TAPPING
+            .tap.interrupted = false,
+            .tap.count       = 0,
+#    endif
+            .keycode = get_alt_repeat_key_keycode(),
+        };
+    }
+
+    // Early return if there is no alternate key defined.
+    if (!registered_record.keycode) {
+        return;
+    }
+
+    if (event->pressed) {
+        update_last_repeat_count(-1);
+        registered_repeat_count = last_repeat_count;
+    }
+
+    // Generate a keyrecord and plumb it into the event pipeline.
+    registered_record.event = *event;
+    processing_repeat_count = registered_repeat_count;
+    process_record(&registered_record);
+    processing_repeat_count = 0;
+}
+
+// Default implementation of get_alt_repeat_key_keycode_user().
+__attribute__((weak)) uint16_t get_alt_repeat_key_keycode_user(uint16_t keycode, uint8_t mods) {
+    return KC_NO;
```

Comment:
If the above change is accepted, this would need to be adjusted too:
```suggestion
    return KC_TRANSPARENT;
```

Reply:
Committed.